src/sentry/services/hybrid_cloud/relocation_export/impl.py

```diff
@@ -0,0 +1,201 @@
+            "requesting_region_name": requesting_region_name,
+            "replying_region_name": replying_region_name,
+            "org_slug": org_slug,
+            "encrypted_contents_size": len(encrypt_with_public_key),
+        }
+        logger.info("SaaS -> SaaS request received in exporting region", extra=logger_data)
+
+        # This task will do the actual work of performing the export and saving it to this regions
+        # "relocation" GCS bucket. It is annotated with the appropriate retry, back-off etc logic
+        # for robustness' sake. The last action performed by this task is to call an instance of
+        # `ControlRelocationExportService.reply_with_export` via a manually-scheduled
+        # `RegionOutbox`, which will handle the task of asynchronously delivering the encrypted,
+        # newly-exported bytes.
+        reply_with_remote_export.apply_async(
+            args=[
+                relocation_uuid,
+                requesting_region_name,
+                replying_region_name,
+                org_slug,
+                encrypt_with_public_key,
+            ]
+        )
+        logger.info("SaaS -> SaaS exporting task scheduled", extra=logger_data)
+
+    def reply_with_export(
+        self,
+        *,
+        relocation_uuid: str,
+        requesting_region_name: str,
+        replying_region_name: str,
+        org_slug: str,
+        encrypted_contents: bytes,
+    ) -> None:
+        from sentry.tasks.relocation import uploading_complete
+
+        with atomic_transaction(
+            using=(
+                router.db_for_write(Relocation),
+                router.db_for_write(RelocationFile),
+                router.db_for_write(File),
+            )
+        ):
+            logger_data = {
+                "uuid": relocation_uuid,
+                "requesting_region_name": requesting_region_name,
+                "replying_region_name": replying_region_name,
+                "org_slug": org_slug,
+                "encrypted_contents_size": len(encrypted_contents),
+            }
+            logger.info("SaaS -> SaaS reply received in triggering region", extra=logger_data)
+
+            try:
+                relocation: Relocation = Relocation.objects.get(uuid=relocation_uuid)
+            except Relocation.DoesNotExist as e:
+                logger.exception("Could not locate Relocation model by UUID: %s", relocation_uuid)
+                capture_exception(e)
+                return
+
+            fp = BytesIO(encrypted_contents)
+            file = File.objects.create(name="raw-relocation-data.tar", type=RELOCATION_FILE_TYPE)
+            file.putfile(fp, blob_size=RELOCATION_BLOB_SIZE, logger=logger)
+            logger.info("SaaS -> SaaS relocation underlying File created", extra=logger_data)
+
+            RelocationFile.objects.create(
+                relocation=relocation,
+                file=file,
+                kind=RelocationFile.Kind.RAW_USER_DATA.value,
+            )
```

Comment:
We'll need a way to make this idempotent. Outboxes can be retried multiple times because of timeouts/connection errors.

Reply:
I've added a small migration, restricting `RelocationFile` to uniqueness per relocation_uuid + relocation_file_kind pair. This means that even if these outbox retries trigger (wasteful, but not otherwise horrible) duplications at other parts of the chain, the successful export response can only be committed at most once in the requesting region. So for example, if each time we manually write an outbox in this chain we end up sending 2 retries, this method would be triggered 8 times for this particular relocation UUID. However, since only one of those files can win the race, the other 7 will result in errors, and no `uploading_complete` being scheduled.

Reply:
As long as the race settles out to the same state and the duplicate attempts aren't counted as errors that sounds good to me.